packages/block-library/src/utils/use-context-data.js

```diff
@@ -0,0 +1,368 @@
+ *
+ * @param {string} templateSlug The template slug to parse.
+ * @return {Object} Object containing taxonomy, termSlug, isAuthor, and authorSlug.
+ */
+export function parseTemplateSlug( templateSlug ) {
+	if ( ! templateSlug ) {
+		return {
+			taxonomy: null,
+			termSlug: null,
+			isAuthor: false,
+			authorSlug: null,
+		};
+	}
+
+	// Check for author patterns
+	// e.g. author, author-john-doe
+	const authorMatches = templateSlug.match( /^(author)$|^author-(.+)$/ );
+	if ( authorMatches ) {
+		return {
+			taxonomy: null,
+			termSlug: null,
+			isAuthor: true,
+			authorSlug: authorMatches[ 2 ] || null,
+		};
+	}
+
+	// Check for taxonomy patterns
+	// e.g. taxonomy-product-category, taxonomy-product-category-electronics
+	if ( templateSlug.startsWith( 'taxonomy-' ) ) {
+		const taxonomyPart = templateSlug.substring( 9 );
+
+		// Find dash positions
+		const dashIndices = [];
+		for ( let i = 0; i < taxonomyPart.length; i++ ) {
+			if ( taxonomyPart[ i ] === '-' ) {
+				dashIndices.push( i );
+			}
+		}
+
+		if ( dashIndices.length > 0 ) {
+			// If there's only one dash, treat it as taxonomy.
+			// If there are multiple dashes, use the last one as the split point.
+			if ( dashIndices.length === 1 ) {
+				// Single dash
+				// e.g. taxonomy-product-category -> taxonomy: "product-category", term: null
+				return {
+					taxonomy: taxonomyPart,
+					termSlug: null,
+					isAuthor: false,
+					authorSlug: null,
+				};
+			}
+
+			// Multiple dashes
+			// e.g. taxonomy-product-category-electronics -> taxonomy: "product-category", term: "electronics"
+			const lastDashIndex = dashIndices[ dashIndices.length - 1 ];
+			const taxonomy = taxonomyPart.substring( 0, lastDashIndex );
+			const termSlug = taxonomyPart.substring( lastDashIndex + 1 );
+
+			return {
+				taxonomy,
+				termSlug,
+				isAuthor: false,
+				authorSlug: null,
+			};
+		}
+
+		// No dashes, so the entire part is the taxonomy
+		return {
+			taxonomy: taxonomyPart,
+			termSlug: null,
+			isAuthor: false,
+			authorSlug: null,
+		};
+	}
+
+	// Check for built-in taxonomy patterns
+	// e.g. category, tag
+	if ( templateSlug === 'category' ) {
+		return {
+			taxonomy: 'category',
+			termSlug: null,
+			isAuthor: false,
+			authorSlug: null,
+		};
+	}
+	if ( templateSlug === 'tag' ) {
+		return {
+			taxonomy: 'post_tag',
+			termSlug: null,
+			isAuthor: false,
+			authorSlug: null,
+		};
+	}
+
+	// Check for specific term patterns for built-in taxonomies
+	// e.g. category-news, tag-featured
+	if ( templateSlug.startsWith( 'category-' ) ) {
+		return {
+			taxonomy: 'category',
+			termSlug: templateSlug.substring( 9 ),
+			isAuthor: false,
+			authorSlug: null,
+		};
+	}
+	if ( templateSlug.startsWith( 'tag-' ) ) {
+		return {
+			taxonomy: 'post_tag',
+			termSlug: templateSlug.substring( 4 ),
+			isAuthor: false,
+			authorSlug: null,
+		};
+	}
+
+	return {
+		taxonomy: null,
+		termSlug: null,
+		isAuthor: false,
+		authorSlug: null,
+	};
```

Comment:
Looks like this object is the same as the one we return when there's no template slug. While we don't use the actual object since we always destructure from it, it might be a potential micro-optimization. Especially considering it's also used in the other parsing functions.